src/components/Patient/PatientRegistration.tsx

```diff
@@ -0,0 +1,1076 @@
+      district: matchedDistrict.id,
+    }));
+
+    if (local_body) {
+      await new Promise((resolve) => setTimeout(resolve, 500));
+      const localBodies = await localBodyQuery.refetch();
+
+      const matchedLocalBody = localBodies.data?.find(
+        (lb) => lb.id === local_body,
+      );
+      if (!matchedLocalBody) return;
+      setForm((f) => ({
+        ...f,
+        local_body: matchedLocalBody.id,
+      }));
+
+      if (ward) {
+        await new Promise((resolve) => setTimeout(resolve, 500));
+        const wards = await wardsQuery.refetch();
+
+        const matchedWard = wards.data?.results.find(
+          (w) => w.id === Number(ward),
+        );
+        if (!matchedWard) return;
+        setForm((f) => ({
+          ...f,
+          ward: matchedWard.id.toString(),
+        }));
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (patientQuery.data) {
+      setForm(patientQuery.data);
+      if (patientQuery.data.year_of_birth && !patientQuery.data.date_of_birth) {
+        setAgeDob("age");
+      }
+      if (
+        patientQuery.data.phone_number ===
+        patientQuery.data.emergency_phone_number
+      )
+        setSamePhoneNumber(true);
+      if (patientQuery.data.address === patientQuery.data.permanent_address)
+        setSameAddress(true);
+      setAddress({
+        state: patientQuery.data.state,
+        district: patientQuery.data.district,
+        local_body: patientQuery.data.local_body,
+        ward: patientQuery.data.ward,
+      });
+    }
+  }, [patientQuery.data]);
+
+  const statesQuery = useQuery({
+    queryKey: ["states"],
+    queryFn: query(routes.statesList),
+  });
+
+  const districtsQuery = useQuery({
+    queryKey: ["districts", form.state],
+    enabled: !!form.state,
+    queryFn: query(routes.getDistrictByState, {
+      pathParams: { id: form.state?.toString() || "" },
+    }),
+  });
+
+  const localBodyQuery = useQuery({
+    queryKey: ["localbodies", form.district],
+    enabled: !!form.district,
+    queryFn: query(routes.getLocalbodyByDistrict, {
+      pathParams: { id: form.district?.toString() || "" },
+    }),
+  });
+
+  const wardsQuery = useQuery({
+    queryKey: ["wards", form.local_body],
+    enabled: !!form.local_body,
+    queryFn: query(routes.getWardByLocalBody, {
+      pathParams: { id: form.local_body?.toString() || "" },
+    }),
+  });
+
+  const handlePincodeChange = async (value: string) => {
+    if (!validatePincode(value)) return;
+    if (form.state && form.district) return;
+
+    const pincodeDetails = await getPincodeDetails(
+      value,
+      careConfig.govDataApiKey,
+    );
+    if (!pincodeDetails) return;
+
+    const matchedState = statesQuery.data?.results?.find((state) => {
+      return includesIgnoreCase(state.name, pincodeDetails.statename);
+    });
+    if (!matchedState) return;
+    setForm((f) => ({
+      ...f,
+      state: matchedState.id,
+    }));
+    await new Promise((resolve) => setTimeout(resolve, 500));
```

Comment:
We should probably have a utility function for loading items as a result of a field being selected

Reply:
I think `useQuery` with a dynamic queryKey would serve the purpose? Or do you mean something else?

Reply:
queryKey dependent on the dependency is sufficient. But if we have queryKey, why are we doing sleep and refetch? 🤔

Reply:
@rithviknishad hack https://github.com/ohcnetwork/care_fe/pull/9400#discussion_r1895771214